Removal of safe resolution [ANT-3566]

a-zakir opened the pull request.

flomnes committed:
Add modeler features 8.1, 8.2 and 8.3 [ANT-3535] (#2902)

- Keep only TS1 for wind, thermal and load other TS are not used (3*40
Mb -> 3*1,5 Mb)
- For 8_1 & 8_2, add checks on values based on hand-made computations
- For 8_3 only check the objective value since time coupling introduces
multiple equivalent solutions.

payetvin committed:
Dynamic groups for thermal and renewable clusters [ANT-1672] (#2579)

This pull request includes several changes to the codebase, focusing on
refactoring, improving conditional logic, and updating versioning. The
most important changes include adding conditional checks to workflows,
removing redundant matrix test functions, replacing custom exceptions
with standard ones, and updating version numbers.

### Workflow Improvements:
*
[`.github/workflows/ubuntu.yml`](diffhunk://#diff-6640e8c668a6deeb426558faaa2b0f535d95814c09a05c044a1df27a008dd544R248):
Added conditional checks to run cucumber tests only if specific
environment variables are set and the job is not cancelled.
[[1]](diffhunk://#diff-6640e8c668a6deeb426558faaa2b0f535d95814c09a05c044a1df27a008dd544R248)
[[2]](diffhunk://#diff-6640e8c668a6deeb426558faaa2b0f535d95814c09a05c044a1df27a008dd544R288)
*
[`.github/workflows/windows-vcpkg.yml`](diffhunk://#diff-8e86046c750ae789851a44583d756b322d7908feaafb4776eb93ac4e331d1d13R249):
Added similar conditional checks for cucumber tests.
[[1]](diffhunk://#diff-8e86046c750ae789851a44583d756b322d7908feaafb4776eb93ac4e331d1d13R249)
[[2]](diffhunk://#diff-8e86046c750ae789851a44583d756b322d7908feaafb4776eb93ac4e331d1d13R289)

### Codebase Simplification:
*
[`src/libs/antares/array/include/antares/array/matrix.h`](diffhunk://#diff-ec412f98c794bbf244613bd015d928e0990628b9a30ef63762d044d69ac586baL544-L585):
Removed multiple matrix test functions that were redundant.
*
[`src/libs/antares/array/matrix.cpp`](diffhunk://#diff-f82f00ff935edcc1c83d0e7e8066fb4d7866840c256dcb5c287cc39ebd80c62eL2-L132):
Removed the corresponding implementations of the matrix test functions.

### Exception Handling:
*
[`src/libs/antares/include/antares/antares/Enum.hxx`](diffhunk://#diff-f4b7f6248f5568080139a2e1caed1b11e131e085113545919655ba34d22b3452R27-L30):
Replaced `AssertionError` with `std::runtime_error` for exception
handling.
[[1]](diffhunk://#diff-f4b7f6248f5568080139a2e1caed1b11e131e085113545919655ba34d22b3452R27-L30)
[[2]](diffhunk://#diff-f4b7f6248f5568080139a2e1caed1b11e131e085113545919655ba34d22b3452L41-R40)
[[3]](diffhunk://#diff-f4b7f6248f5568080139a2e1caed1b11e131e085113545919655ba34d22b3452L54-R53)
*
[`src/libs/antares/study/UnfeasibleProblemBehavior.cpp`](diffhunk://#diff-22ee2f6b03856b19380aded0fb1e8ab2351478ae5c5ba6907078eb50678ec92fL44-R43):
Updated exception handling to use `std::runtime_error` instead of
`AssertionError`.
[[1]](diffhunk://#diff-22ee2f6b03856b19380aded0fb1e8ab2351478ae5c5ba6907078eb50678ec92fL44-R43)
[[2]](diffhunk://#diff-22ee2f6b03856b19380aded0fb1e8ab2351478ae5c5ba6907078eb50678ec92fL62-R61)
[[3]](diffhunk://#diff-22ee2f6b03856b19380aded0fb1e8ab2351478ae5c5ba6907078eb50678ec92fL80-R79)
[[4]](diffhunk://#diff-22ee2f6b03856b19380aded0fb1e8ab2351478ae5c5ba6907078eb50678ec92fL100-R99)

### Version Updates:
*
[`simtest.json`](diffhunk://#diff-c2088552287e6bf08a1bf79787828d4e59d6a9b1010156e943cd7196e2365be6L2-R2):
Updated the version from "v9.2.0i" to "v9.2.0j".
*
[`src/CMakeLists.txt`](diffhunk://#diff-148715d6ea0c0ea0a346af3f6bd610d010d490eca35ac6a9b408748f7ca9e3f4L10-R10):
Updated the release candidate number from 9 to 10.

### Refactoring:
*
[`src/libs/antares/study/include/antares/study/parts/common/cluster.h`](diffhunk://#diff-6632e6418938cf399ed6b48a78bdcc573dd0538dafcd159f7881b864fb65980bL57-R63):
Changed `ClusterName` type from a custom type to `std::string`.
[[1]](diffhunk://#diff-6632e6418938cf399ed6b48a78bdcc573dd0538dafcd159f7881b864fb65980bL57-R63)
[[2]](diffhunk://#diff-6632e6418938cf399ed6b48a78bdcc573dd0538dafcd159f7881b864fb65980bL138-R140)
*
[`src/libs/antares/study/include/antares/study/parts/common/cluster_list.h`](diffhunk://#diff-8204a7789a02a029e28598b45a487326de846f7df1487f61d05d34af60538108L67-R69):
Updated methods to use `std::string` for cluster IDs.
[[1]](diffhunk://#diff-8204a7789a02a029e28598b45a487326de846f7df1487f61d05d34af60538108L67-R69)
[[2]](diffhunk://#diff-8204a7789a02a029e28598b45a487326de846f7df1487f61d05d34af60538108L86-R93)

These changes improve the maintainability and readability of the code
and ensure better conditional handling in workflows.

---------

Co-authored-by: GitHub Actions <actions@github.com>
Co-authored-by: Florian OMNES <florian.omnes@rte-france.com>
Co-authored-by: Florian OMNES <26088210+flomnes@users.noreply.github.com>
Co-authored-by: Jason Maréchal <45510813+JasonMarechal25@users.noreply.github.com>

a-zakir committed:
remove 'Safe resolution'

a-zakir committed:
remove 'Safe resolution'

flomnes commented on src/solver/optimisation/opt_appel_solveur_lineaire.cpp:
```diff
@@ -246,7 +241,7 @@ static SimplexResult OPT_TryToCallSimplex(const SingleOptimOptions& options,
         }
     }
-
+    // TODO could this condition be true?
```
Yes, at the 1st resolution see #2868

JasonMarechal25 committed:
[ANT-2034] InMemory modeler test (#2827)

Refactoring to allow in memory modeler test.

- Move _modeler-lib_ outside _modeler_ exe scope
- _modeler-lib_ now expose a _Modeler_ and some interfaces
- _modeler_ exe depends on modeler-lib for _Modeler_ and _IFileWriter_
interface
- _load-modeler-files_ depends on _modeler-libs_ for _ILoader_ interface

---------

Co-authored-by: Florian Omnès <florian.omnes@rte-france.com>
Co-authored-by: Peter Mitri <peter.mitri@rte-france.com>

flomnes committed:
Merge remote-tracking branch 'github/develop' into feature/delete-safe-resolution

flomnes committed:
Revert "Merge remote-tracking branch 'github/develop' into feature/delete-safe-resolution"

This reverts commit 594f4781c1f7e524230b380bc2438aab4d226fe7, reversing
changes made to 85a7330b6895649d1b66d2225104b06d9526bb16.

a-zakir committed:
fix seg fault

a-zakir committed:
Merge remote-tracking branch 'origin/feature/delete-safe-resolution' into feature/delete-safe-resolution

a-zakir committed:
update
